pkg/controller/v1beta1/inferenceservice/reconcilers/ingress/httproute_reconciler.go

```diff
@@ -0,0 +1,804 @@
+	}
+
+	// Add isvc name and namespace headers
+	filters := []gatewayapiv1.HTTPRouteFilter{addIsvcHeaders(isvc.Name, isvc.Namespace)}
+
+	explainerHost, err := GenerateDomainName(explainerName, isvc.ObjectMeta, ingressConfig)
+	if err != nil {
+		return nil, fmt.Errorf("failed to generate explainer ingress host: %w", err)
+	}
+	allowedHosts = append(allowedHosts, gatewayapiv1.Hostname(explainerHost))
+
+	// Add explainer host rules
+	routeMatch := []gatewayapiv1.HTTPRouteMatch{createHTTPRouteMatch(constants.FallbackPrefix())}
+	httpRouteRules = append(httpRouteRules, createHTTPRouteRule(routeMatch, filters, explainerName, isvc.Namespace,
+		constants.CommonDefaultHttpPort))
+
+	annotations := utils.Filter(isvc.Annotations, func(key string) bool {
+		return !utils.Includes(constants.ServiceAnnotationDisallowedList, key)
+	})
+	gatewaySlice := strings.Split(ingressConfig.KserveIngressGateway, "/")
+	httpRoute := gatewayapiv1.HTTPRoute{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        constants.ExplainerServiceName(isvc.Name),
+			Namespace:   isvc.Namespace,
+			Annotations: annotations,
+			Labels:      isvc.Labels,
+		},
+		Spec: gatewayapiv1.HTTPRouteSpec{
+			Hostnames: allowedHosts,
+			Rules:     httpRouteRules,
+			CommonRouteSpec: gatewayapiv1.CommonRouteSpec{
+				ParentRefs: []gatewayapiv1.ParentReference{
+					{
+						Group:     (*gatewayapiv1.Group)(&gatewayapiv1.GroupVersion.Group),
+						Kind:      (*gatewayapiv1.Kind)(utils.ToPointer(constants.GatewayKind)),
+						Namespace: (*gatewayapiv1.Namespace)(&gatewaySlice[0]),
+						Name:      gatewayapiv1.ObjectName(gatewaySlice[1]),
+					},
+				},
+			},
+		},
+	}
+	return &httpRoute, nil
+}
+
+func createRawTopLevelHTTPRoute(isvc *v1beta1.InferenceService, ingressConfig *v1beta1.IngressConfig,
+	client client.Client) (*gatewayapiv1.HTTPRoute, error) {
+	var httpRouteRules []gatewayapiv1.HTTPRouteRule
+	var allowedHosts []gatewayapiv1.Hostname
+	additionalHosts := ingressConfig.AdditionalIngressDomains
+
+	if !isvc.Status.IsConditionReady(v1beta1.PredictorReady) {
+		isvc.Status.SetCondition(v1beta1.IngressReady, &apis.Condition{
+			Type:   v1beta1.IngressReady,
+			Status: corev1.ConditionFalse,
+			Reason: "Predictor ingress not created",
+		})
+		return nil, nil
+	}
+	existing := &corev1.Service{}
+	predictorName := constants.PredictorServiceName(isvc.Name)
+	transformerName := constants.TransformerServiceName(isvc.Name)
+	explainerName := constants.ExplainerServiceName(isvc.Name)
+	// Check if existing predictor service name has default suffix
+	err := client.Get(context.TODO(), types.NamespacedName{Name: constants.DefaultPredictorServiceName(isvc.Name), Namespace: isvc.Namespace}, existing)
+	if err == nil {
+		// Found existing predictor service with default suffix
+		predictorName = constants.DefaultPredictorServiceName(isvc.Name)
+		transformerName = constants.DefaultTransformerServiceName(isvc.Name)
+		explainerName = constants.DefaultExplainerServiceName(isvc.Name)
+	}
```

Comment:
We have deprecated the default name for a while now, maybe it is safe to assume that this does not exist anymore if it can make code cleaner

Reply:
I think we can do it in a separate PR. There are other places which also needs to be changed